src/gc/collection.rs

```diff
@@ -139,10 +135,13 @@ async unsafe fn process_mutation_buffer(
 ) {
     // It is very important that we do not delay any mutations that
     // have occurred at this point by an extra epoch.
-    let to_recv = mutation_buffer_rx.len();
+    let to_recv = mutation_buffer_rx
+        .len()
+        .min(MIN_MUTATIONS_PER_EPOCH)
+        .max(MAX_MUTATIONS_PER_EPOCH);
```

Comment:
Our goal here is that if X mutations occurred while processing the last epoch, that _at least_ X mutations are processed in the next epoch. This code does not do that unfortunately. Let's say that there were 10,001 mutations in the last epoch:

`mutation_buffer_rx.len() = 10,001`
`min(mutation_buffer.len(), 10) = 10`
`max(min(mutation_buffer.len(), 10) = 10, 10,000) = 10,000`

so, `to_recv` will always be `10,000` by the end of this expression, no matter if it less than min or less that max. 

I understand here you intend is to `clamp`, but again it's absolutely necessary that _at least_ `mutation_buffer_rx.len()` items are processed, so clamping would be incorrect.